### DISCUSSION TITLE
Transfer rowData to splitbutton for actions in dataTable

### DISCUSSION THREAD

**Discussion by Weicai-Ling** [primefaces/community#350] [PrimeVue] [answered]
Who could please help to advise how to transfer rowData to splitbutton in each rows in dataTable ?

**Comment by Weicai-Ling** [comment]
I solved this issue by displaying {{ rowData.data.name }} . Finally I know the rowData in <template #body="rowData" > is the object of the parent of data object.

**Reply by FlipWarthog** [reply]
Can you mark this as the answer?

**Reply by Weicai-Ling** [reply]
It is not the perfect answer.

**Comment by tugcekucukoglu** [comment]
This is generally how the body slot works. It may be necessary to re-check the data coming to the body template.

**Reply by Weicai-Ling** [reply]
Thanks @tugcekucukoglu for reply. I working on the issue for several days. Really appreciate if you can advise some words on my issue. 

I asked chat.openai with a solution in options API, I changed it to Composition API below:  

<script setup>
const data = ref([
    { name: 'Item 1', id: 1 },
    { name: 'Item 2', id: 2 },
    { name: 'Item 3', id: 3 },
    { name: 'Item 4', id: 4 }
]);

const splitButtonItems = ref([
    {
        label: 'View',
        icon: 'pi pi-search',
        command: (rowData) => { viewItem(rowData); },
    },
    {
        label: 'Edit',
        icon: 'pi pi-pencil',
        command: (rowData) => { editItem(rowData); },
    },
    {
        label: 'Delete',
        icon: 'pi pi-trash',
        command: (rowData) => { deleteItem(rowData); }
    },
]);

const viewItem = (rowData) => {
    console.log('View item:', rowData);
};

const editItem = (rowData) => {
    console.log('Edit item:', rowData);
};

const deleteItem = (rowData) => {
    console.log('Delete item:', rowData);
};

const handleSplitButtonClick = (rowData) => {
    console.log('handle default click:', rowData);
};

</script>

<template>
                <DataTable :value="data">
                    <Column field="name" header="Name"></Column>
                    <Column field="actions" header="Actions">
                        <template #body="rowData">
                            <SplitButton label="Actions" :model="splitButtonItems" @click="handleSplitButtonClick(rowData)">
                            </SplitButton>
                        </template>
                    </Column>
                </DataTable>
</template>

when I click the Edit dropdown item, the console log is : 

Edit item: {originalEvent: PointerEvent, item: Proxy(Object)}item: Proxy(Object) {label: 'Edit', icon: 'pi pi-pencil', command: ƒ}[[Handler]]: Object[[Target]]: Object[[IsRevoked]]: falseoriginalEvent: PointerEvent {isTrusted: true, _vts: 1699016920172, pointerId: 1, width: 1, height: 1, …}[[Prototype]]: Object

Really appreciate if you can help ! 

Forrest Weicai Ling

**Reply by Weicai-Ling** [reply]
`                <DataTable :value="data">
                    <Column field="name" header="Name"></Column>
                    <Column field="actions" header="Actions">
                        <template #body="rowData">
                            <SplitButton label="Actions" :model="splitButtonItems" @click="handleSplitButtonClick(rowData)">
                            </SplitButton>
                        </template>
                    </Column>
                </DataTable>`

**Comment by Weicai-Ling** [comment]
Are there any suggestions/example code for embedded SplitButton in DataTable for selected action on corresponding row?

**Reply by Weicai-Ling** [reply]
I asked ChatGPT for generating below codes, but It does not work. Who could please help advise on SplitButton in DataTable. Really appreciate if any experts can help me on this problem. 
 

```
<template>
  <div>
    <DataTable :value="data">
      <Column field="id" header="ID"></Column>
      <Column field="name" header="Name"></Column>
      <Column field="description" header="Description"></Column>
      <Column header="Actions">
        <template #body="{ rowData }">
          <SplitButton :model="splitButtonItems" @click="handleAction(rowData)"></SplitButton>
        </template>
      </Column>
    </DataTable>
  </div>
</template>

<script setup>
import { ref } from 'vue';
import { DataTable, Column, SplitButton } from 'primevue/datatable';

const data = ref([
  { id: 1, name: 'Item 1', description: 'Description for Item 1' },
  { id: 2, name: 'Item 2', description: 'Description for Item 2' },
  { id: 3, name: 'Item 3', description: 'Description for Item 3' },
]);

const splitButtonItems = [
  {
    label: 'View',
    icon: 'pi pi-search',
    command: 'view',
  },
  {
    label: 'Edit',
    icon: 'pi pi-pencil',
    command: 'edit',
  },
  {
    label: 'Delete',
    icon: 'pi pi-trash',
    command: 'delete',
  },
];

const handleAction = (rowData) => {
  switch (rowData.command) {
    case 'view':
      // Implement view action
      break;
    case 'edit':
      // Implement edit action
      break;
    case 'delete':
      // Implement delete action
      break;
    default:
      break;
  }
};
</script>
```

**Comment by aasiph** [comment, marked as answer]
Check this https://github.com/orgs/primefaces/discussions/834
Maybe this is what you wanted..
```javascript
function editAction(data) {
 \\you get the row data here
}
```
```html
<Column  header="Action" >
      <template #body="{ data }">
          <SplitButton
              icon="pi pi-ellipsis-v"
              text
              :model="[
                  { label: 'Edit', icon: 'pi pi-fw pi-pencil', command: () => editAction(data) },
                  { label: 'Delete', icon: 'pi pi-fw pi-trash', command: () => deleteAction(data) }
              ]"
          ></SplitButton>
      </template>
  </Column>
```

**Reply by Weicai-Ling** [reply]
@aasiph Thanks ! Happy New Year 2024 !
